var/spack/repos/builtin/packages/psrchive/package.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class Psrchive(AutotoolsPackage):
+    """PSRCHIVE is a library for the analysis of pulsar astronomical data.
+    PSRCHIVE is an Open Source C++ development library for the analysis of
+    pulsar astronomical data.  It implements an extensive range of algorithms
+    for use in pulsar timing, scintillation studies, polarimetric calibration,
+    single-pulse work, RFI mitigation, etc. These tools are utilized by a
+    powerful suite of user-end programs that come with the library."""
+
+    homepage = "http://psrchive.sourceforge.net/"
+    url = "https://sourceforge.net/projects/psrchive/files/psrchive/2022-05-14/psrchive-2022-05-14.tar.gz/download"
+    git = "https://git.code.sf.net/p/psrchive/code.git"
+
+    version(
+        "2022-05-14", sha256="4d25609837cba1be244fa8adc8f105afe31972f2650bc0b90438862cf35395e1"
+    )
+
+    # version specified in
+    # https://github.com/lwa-project/pulsar/blob/master/SoftwareStack.md
+    # as of Nov 23 2022
+    version("10-17-2020", commit="ca12b4a279f3d4adcca223508116d9d270df8cc6")
+
+    variant("mpi", default=True, description="Compile with MPI")
+    variant("mkl", default=False, description="Compile with MKL")
+    variant("armadillo", default=False, description="Compile with armadillo")
+    variant("cfitsio", default=False, description="Compile with cfitsio")
+    variant("eigen", default=False, description="Compile with eigen")
+    variant("xml", default=False, description="Compile with libxml2")
+    variant("x11", default=False, description="Compile with X11")
+    variant("qt", default=False, description="Compile with QT")
+
+    depends_on("autoconf", type="build")
+    depends_on("automake", type="build")
+    depends_on("libtool", type="build")
+    depends_on("psrcat")
+    depends_on("epsic")
+    depends_on("tempo")
+    depends_on("fftw@3:")
+    depends_on("python")
+    depends_on("pgplot")
+
+    depends_on("mpi", when="+mpi")
+    depends_on("intel-mkl", when="+mkl")
```

Comment:
```suggestion
    depends_on("mkl", when="+mkl")
```

There are multiple providers for `mkl` among the different intel packages/bundles, this way it will allow Spack to use any of them to satisfy the dependency.